### PR TITLE
[Merged by Bors] - feat(Topology/Order): Sierpiński space classifies open sets of Scott topology

### PR DESCRIPTION
**Pull request by mans0954**
This PR shows that the Sierpiński topology coincides with the upper topology on `Prop`, and hence with the Scott topology (since `Prop` is a complete linear order).

This observation reveals that a subset of a preorder is open in the Scott topology if and only if its characteristic function is Scott continuous.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mans0954** [comment]
CC: @YaelDillies

File: Mathlib/Topology/Order/LowerUpperTopology.lean [hunk 1]
```diff
@@ -509,3 +509,27 @@ lemma isLower_orderDual [Preorder α] [TopologicalSpace α] : IsLower αᵒᵈ 
...
+        rw [← ha]
+        by_cases h : a = True
+        · apply Or.inl
+          rw [h, Iic_True_compl]
+        · apply Or.inr
+          rw [← Iic_False_compl]
+          aesop)
+      (fun _ hs => by rw [← Iic_False_compl, ← Iic_True_compl] at hs; aesop)
+    rw [Topology.upper, e1, generateFrom_insert_empty]
+    rfl
```
Comment (YaelDillies) [inline Mathlib/Topology/Order/LowerUpperTopology.lean:535]:
```suggestion
/-- The Sierpiński topology on `Prop` is the upper topology -/
instance : IsUpper Prop where
  topology_eq_upperTopology := by
    rw [Topology.upper, sierpinskiSpace, ← generateFrom_insert_empty]
    congr
    exact le_antisymm
      (fun _ hs ↦ by rw [← Iic_False_compl, ← Iic_True_compl] at hs; aesop)
      (by rintro _ ⟨a, rfl⟩; by_cases a <;> aesop (add simp [Ioi, lt_iff_le_not_le]))
```

File: Mathlib/Topology/Order/LowerUpperTopology.lean [hunk 2]
```diff
@@ -509,3 +509,27 @@ lemma isLower_orderDual [Preorder α] [TopologicalSpace α] : IsLower αᵒᵈ 
...
 
 end Topology
+
+lemma Iic_False_compl : (Iic False)ᶜ = {True} := by
+  rw [compl_eq_comm, Prop.compl_singleton, not_true_eq_false]
+  aesop
+
+lemma Iic_True_compl : (Iic True)ᶜ = ∅ := by
+  rw [compl_eq_comm, compl_empty]
+  aesop
```
Comment (YaelDillies) [inline Mathlib/Topology/Order/LowerUpperTopology.lean:519]: These two lemmas aren't about topology. Can they move earlier?

**Review by YaelDillies — APPROVED** [review]
I pushed a golf. The rest looks good

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**